src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -1436,35 +1437,39 @@ void QGIViewDimension::resetArrows() const
     aHead2->setFlipped(false);
 }
 
+// NOLINTNEXTLINE
 void QGIViewDimension::drawArrows(int count, const Base::Vector2d positions[], double angles[],
                                   bool flipped, bool forcePoint) const
 {
-    const int arrowCount = 2;
-    QGIArrow* arrows[arrowCount] = {aHead1, aHead2};
-
+    constexpr int arrowCount{2};
+    QGIArrow* arrows[arrowCount] = {aHead1, aHead2};    // NOLINT
     arrowPositionsToFeature(positions);
 
     for (int i = 0; i < arrowCount; ++i) {
-        QGIArrow* arrow = arrows[i];
+        QGIArrow* arrow = arrows[i];                    // NOLINT
 
         if (positions && angles) {
-            arrow->setPos(toQtGui(positions[i]));
-            arrow->setDirection(toQtRad(angles[i]));
+            arrow->setPos(toQtGui(positions[i]));       // NOLINT
+            arrow->setDirection(toQtRad(angles[i]));    // NOLINT
         }
 
         if (i >= count) {
             arrow->hide();
             continue;
         }
 
-        // some dimensions must use point ends (area). The point style is 3.
-        arrow->setStyle(forcePoint ? 3 : QGIArrow::getPrefArrowStyle());
         auto vp = static_cast<ViewProviderDimension*>(getViewProvider(getViewObject()));
+        if (!vp) {
+            return;
+        }
+
+        // some dimensions must use point ends (area). The point style is 3.
+        arrow->setStyle(forcePoint ? ArrowType::DOT : static_cast<ArrowType>(vp->ArrowStyle.getValue()));
```

Comment:
```suggestion
        arrow->setStyle(forcePoint ? ArrowType::DOT : static_cast<ArrowType>(vp->ArrowStyle.getValue()));
```
Since we use enums now